### PR TITLE
Use exit code to indicate check problems

### PR DESCRIPTION
**Pull request by maio**
Exit with code 1 if there is a file with incorrect formatting.
Exit with code 2 if there were some errors.

Useful if you want to perform some action depending on check
results (e.g. trigger build failure).

### CODE REVIEW

File: lein-cljfmt/src/leiningen/cljfmt.clj [hunk 1]
```diff
@@ -84,6 +84,12 @@
       (main/warn path "has incorrect formatting")
       (main/warn diff))))
 
+(defn exit [counts]
+  (when-not (zero? (:incorrect counts 0))
+    (leiningen.core.main/exit 1))
```
Comment (weavejester) [inline lein-cljfmt/src/leiningen/cljfmt.clj]: Why not `(main/exit 1)` here?

File: lein-cljfmt/src/leiningen/cljfmt.clj [hunk 2]
```diff
@@ -105,15 +111,16 @@
...
-   (->> (cons path paths)
-        (transduce
-         (comp (mapcat (partial find-files project))
-               (map (partial check-one project))
-               (map (fn [status]
-                      (print-file-status project status)
-                      (:counts status))))
-         (completing merge-counts))
-        (print-final-count))))
+   (let [counts (->> (cons path paths)
```
Comment (weavejester) [inline lein-cljfmt/src/leiningen/cljfmt.clj]: You could factor out the `->>` part and just put `(cons path paths)` on the end of the transduce.

**Comment by maio** [comment]
@weavejester thanks for comments. They should be fixed now.

**Comment by weavejester** [comment]
Perfect, thanks :)

**Comment by weavejester** [comment]
Just tested it and it all works, but I notice that if there are errors *and* incorrect formatting, it returns an error code of 1. I think errors are more important, so it should return 2 in that case. Do you agree?

**Comment by maio** [comment]
:+1:

**Comment by weavejester** [comment]
Then if you swap the `when-not` statements in the `exit` function, I think it will be ready to merge.

**Comment by maio** [comment]
It should be all fixed now.

**Comment by weavejester** [comment]
Perfect, thanks!

**Pull request merged**